src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -309,49 +309,46 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
 
     /// Get the metadata for a target in a specific profile
     pub fn target_metadata(&self, unit: &Unit) -> Option<Metadata> {
```

Comment:
I think this function should always return `Some` now, right? It may be useful to just return `Metadata` here and then deal with the consequences elsewhere as they arise.